### PR TITLE
Uploading e2e.test in ci-build-kubernetes.sh

### PR DESCRIPTION
**Pull request by marosset**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
I noticed that windows PR tests that use https://github.com/kubernetes-sigs/windows-testing/blob/master/capz/run-capz-e2e.sh do not use a version of e2e.test build from the PR branch/commit.
This PR uploads e2e.test along with other linux binaries so we can download it during PR jobs.
I also improved logging

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [X] cherry-pick candidate 

We should cherry-pick this into release-1.11

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Upload e2e.test in ci-build-kubernetes.sh
```

### CODE REVIEW

File: scripts/ci-build-kubernetes.sh [hunk 1]
```diff
@@ -112,7 +112,10 @@ main() {
             docker tag "${OLD_IMAGE_URL}" "${NEW_IMAGE_URL}" && docker push "${NEW_IMAGE_URL}"
         done
 
+        echo "Uploading binaries to Azure storage container ${JOB_NAME}"
+
         for BINARY in "${BINARIES[@]}"; do
+            echo "uploading ${KUBE_GIT_VERSION}/bin/linux/amd64/${BINARY}"
```
Comment (jackfrancis) [inline scripts/ci-build-kubernetes.sh:118]: nit: slight preference for assigning `${KUBE_GIT_VERSION}/bin/linux/amd64/${BINARY}` to a var for re-use

Comment (marosset) [reply scripts/ci-build-kubernetes.sh]: updated

**Comment by CecileRobertMichon** [comment]
started the job to validate this scenario ^

/hold

feel free to remove the hold if the job passes and everything looks good

/lgtm
/approve

**Comment by marosset** [comment]
/hold cancel
All CI jobs passed

**Pull request merged**